EIPS/eip-225.md

```diff
@@ -0,0 +1,474 @@
+  }, {
+    // Two signers, actually needing mutual consent to drop either of them (fulfilled)
+    signers: []string{"A", "B"},
+    blocks:  []block{
+      {signer: "A", voted: "B", auth: false},
+      {signer: "B", voted: "B", auth: false},
+    },
+    results: []string{"A"},
+  }, {
+    // Three signers, two of them deciding to drop the third
+    signers: []string{"A", "B", "C"},
+    blocks:  []block{
+      {signer: "A", voted: "C", auth: false},
+      {signer: "B", voted: "C", auth: false},
+    },
+    results: []string{"A", "B"},
+  }, {
+    // Four signers, consensus of two not being enough to drop anyone
+    signers: []string{"A", "B", "C", "D"},
+    blocks:  []block{
+      {signer: "A", voted: "C", auth: false},
+      {signer: "B", voted: "C", auth: false},
+    },
+    results: []string{"A", "B", "C", "D"},
+  }, {
+    // Four signers, consensus of three already being enough to drop someone
+    signers: []string{"A", "B", "C", "D"},
+    blocks:  []block{
+      {signer: "A", voted: "D", auth: false},
+      {signer: "B", voted: "D", auth: false},
+      {signer: "C", voted: "D", auth: false},
+    },
+    results: []string{"A", "B", "C"},
+  }, {
+    // Authorizations are counted once per signer per target
+    signers: []string{"A", "B"},
+    blocks:  []block{
+      {signer: "A", voted: "C", auth: true},
+      {signer: "B"},
+      {signer: "A", voted: "C", auth: true},
+      {signer: "B"},
+      {signer: "A", voted: "C", auth: true},
+    },
+    results: []string{"A", "B"},
+  }, {
+    // Authorizing multiple accounts concurrently is permitted
+    signers: []string{"A", "B"},
+    blocks:  []block{
+      {signer: "A", voted: "C", auth: true},
+      {signer: "B"},
+      {signer: "A", voted: "D", auth: true},
+      {signer: "B"},
+      {signer: "A"},
+      {signer: "B", voted: "D", auth: true},
+      {signer: "A"},
+      {signer: "B", voted: "C", auth: true},
+    },
+    results: []string{"A", "B", "C", "D"},
+  }, {
+    // Deauthorizations are counted once per signer per target
+    signers: []string{"A", "B"},
+    blocks:  []block{
+      {signer: "A", voted: "B", auth: false},
+      {signer: "B"},
+      {signer: "A", voted: "B", auth: false},
+      {signer: "B"},
+      {signer: "A", voted: "B", auth: false},
+    },
+    results: []string{"A", "B"},
+  }, {
+    // Deauthorizing multiple accounts concurrently is permitted
+    signers: []string{"A", "B", "C", "D"},
+    blocks:  []block{
+      {signer: "A", voted: "C", auth: false},
+      {signer: "B"},
+      {signer: "C"},
+      {signer: "A", voted: "D", auth: false},
+      {signer: "B"},
+      {signer: "C"},
+      {signer: "A"},
+      {signer: "B", voted: "D", auth: false},
+      {signer: "C", voted: "D", auth: false},
+      {signer: "A"},
+      {signer: "B", voted: "C", auth: false},
+    },
+    results: []string{"A", "B"},
+  }, {
+    // Votes from deauthorized signers are discarded immediately (deauth votes)
```

Comment:
I don't understand the difference between `Votes from deauthorized signers are discarded immediately (deauth votes)` and `Votes from deauthorized signers are discarded immediately (auth votes)` the test cases are identical?

Both clearly has `auth = false` i.e, voting to remove!

Reply:
Good catch, copy paste error. Fixed in the upstream test code https://github.com/ethereum/go-ethereum/pull/19222 and updated this spec too.